src/utils/hooks/useUpdateAssignment.tsx

```diff
@@ -0,0 +1,86 @@
+import { AssignmentDetails, FullStructure, SectionAssignee } from '../../utils/types'
+import {
+  ReviewAssignmentStatus,
+  UpdateReviewAssignmentMutation,
+  useUpdateReviewAssignmentMutation,
+} from '../../utils/generated/graphql'
+import { FetchResult } from '@apollo/client'
+
+const useUpdateAssignment = ({ fullStructure }: { fullStructure: FullStructure }) => {
+  const [updateReviewAssignment] = useUpdateReviewAssignmentMutation()
+
+  const submitAssignments = async (
+    assignedSections: SectionAssignee | null,
+    assignmentsFiltered: AssignmentDetails[]
+  ) => {
+    // NOTE: If "assignedSections ==  null", then ALL sections will be
+    // UNASSIGNED from the review assignments in assignmentsFiltered
+    const results: Promise<
+      FetchResult<UpdateReviewAssignmentMutation, Record<string, any>, Record<string, any>>
+    >[] = []
+    // Deduce which review assignments need to change and create patches, then
+    // mutate one by one
+    try {
+      assignmentsFiltered.forEach((assignment) => {
+        if (isChanging(assignment.reviewer.id, assignedSections)) {
+          const assignmentPatch = createAssignmentPatch(assignment, assignedSections)
+          results.push(
+            updateReviewAssignment({
+              variables: {
+                assignmentId: assignment.id,
+                assignmentPatch,
+              },
+            })
+          )
+        }
+      })
+      await Promise.all(results)
+    } catch (err) {
+      throw new Error('Assignment update error')
+    }
+    fullStructure.reload()
+  }
+
+  return { submitAssignments }
+}
+
+const isChanging = (reviewerId: number, assignedSections: SectionAssignee | null) => {
+  if (!assignedSections) return true
+  else
+    return !!Object.values(assignedSections).find(
+      (section) => section.newAssignee === reviewerId || section.previousAssignee === reviewerId
+    )
+}
+interface AssignmentPatch {
+  status: ReviewAssignmentStatus
+  assignedSections?: string[]
+}
+
+const createAssignmentPatch = (
+  assignment: AssignmentDetails,
+  assignedSections: SectionAssignee | null
+) => {
+  if (!assignedSections) return { status: ReviewAssignmentStatus.Available }
```

Comment:
Is this one to unassign a complete review_assignment? And is it only setting the status to `Available` since the back-end will make sure to set the array to empty/null?

Reply:
This line is just a guard for the case when "assignedSections" is null. Originally I had it that assignedSections couldn't be `null`, but when I was doing the unassignment, I needed a way to unassign from all sections without having to add a bunch of additional logic. 

If you look at "AssignmentSectionRow", you'll see my commented out code that would unassign one section at a time. If we went that way, we wouldn't need this line here either.